### PR TITLE
Make sure that the file is closed on deletion

### PR DESCRIPTION
**Pull request by oruebel**
## Motivation

This is to ensure that files are closed on deletion of the HDMFIO object.

## Checklist

- [x] Did you update CHANGELOG.md with your changes?
- [X] Have you checked our [Contributing](https://github.com/hdmf-dev/hdmf/blob/dev/docs/CONTRIBUTING.rst) document?
- [X] Have you ensured the PR clearly describes the problem and the solution?
- [X] Is your contribution compliant with our coding style? This can be checked running `flake8` from the source directory.
- [X] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/hdmf-dev/hdmf/pulls) for the same change?
- [X] Have you included the relevant issue number using "Fix #XXX" notation where XXX is the issue number? By including "Fix #XXX" you allow GitHub to close issue #XXX when the PR is merged.

### CODE REVIEW

File: src/hdmf/backends/io.py [hunk 1]
```diff
@@ -132,3 +132,6 @@ def __enter__(self):
 
     def __exit__(self, type, value, traceback):
         self.close()
+
```
Comment (oruebel) [inline src/hdmf/backends/io.py]:
```suggestion

```

File: src/hdmf/backends/io.py [hunk 2]
```diff
@@ -132,3 +132,5 @@ def __enter__(self):
 
     def __exit__(self, type, value, traceback):
         self.close()
+    def __del__(self):
```
Comment (oruebel) [inline src/hdmf/backends/io.py:136]:
```suggestion

    def __del__(self):
```

**Review by rly — DISMISSED** [review]

**Comment by rly** [comment]
Can you update the changelog?

**Comment by oruebel** [comment]
> Can you update the changelog?

Done

**Review by rly — DISMISSED** [review]

**Comment by oruebel** [comment]
@rly can you please re-approve. The pipeline to check external links was failing due to a bad link for codecov. Since I was going through the linkchecker anyways, I also fixed a bunch of links that were being redirected to point to the correct new location.

**Review by rly — APPROVED** [review]

**Pull request merged**